test/__runner__/browser/index.js

```
@@ -124,23 +117,25 @@ function generate(options) {
 
 function runTests(options) {
     return generate(options).then(generated => {
-        console.log(`Running ${generated.url} using mocha-phantomjs...`);
-        var mochaPhantomJSOptions = { useColors: true };
-
-        if (shouldCover) {
-            mochaPhantomJSOptions.hooks = 'mocha-phantomjs-istanbul';
-            console.log(options.testsFile)
-            mochaPhantomJSOptions.coverageFile = getCoverageFile(options.testsFile);
-        }
-
-        return spawn(phantomjsBinPath, [mochaPhantomJSCommand, generated.url, 'spec', JSON.stringify(mochaPhantomJSOptions)], {
-            stdio: 'inherit'
+        var cleanup = JSDOM(fs.readFileSync(generated.url, 'utf-8'), {
+            url: 'file://' + generated.url,
+            runScripts: 'dangerously',
+            pretendToBeVisual: true,
+            resources: 'usable'
         });
-    });
-}
 
-function getCoverageFile(testsFile) {
-    return './.nyc_output/' + md5Hex(testsFile) + '.json';
+        return new Promise(function (resolve, reject) {
+            window.addEventListener('load', function () {
+                var runner = window.MOCHA_RUNNER;
+                runner.on('end', function () {
+                    cleanup();
+                    runner.stats.failures.length
+                        ? reject(new Error(''))
```

Should we report the errors? Perhaps `.join` them?